06_deploy_bootstrap_vm.sh

```diff
@@ -19,19 +19,20 @@ sleep 10
 
 INFRA_ID=$(jq -r .infraID ocp/metadata.json)
 
-while ! domain_net_ip ${INFRA_ID}-bootstrap baremetal; do
-  echo "Waiting for ${INFRA_ID}-bootstrap interface to become active.."
-  sleep 10
-done
-
 # NOTE: This is equivalent to the external API DNS record pointing the API to the API VIP
-IP=$(domain_net_ip ${INFRA_ID}-bootstrap baremetal)
-export API_VIP=$(dig +noall +answer "api.${CLUSTER_DOMAIN}" @$(network_ip baremetal) | awk '{print $NF}')
-echo "address=/api.${CLUSTER_DOMAIN}/${API_VIP}" | sudo tee /etc/NetworkManager/dnsmasq.d/openshift.conf
-sudo systemctl reload NetworkManager
+if [ "$NODES_PLATFORM" == 'libvirt' ]; then
+  export API_VIP=$(dig +noall +answer "api.${CLUSTER_DOMAIN}" @$(network_ip baremetal) | awk '{print $NF}')
+  echo "address=/api.${CLUSTER_DOMAIN}/${API_VIP}" | sudo tee /etc/NetworkManager/dnsmasq.d/openshift.conf
+  sudo systemctl reload NetworkManager
+else
+  export API_VIP=$(dig +short "api.${CLUSTER_DOMAIN}")
+fi
 
 # Wait for ssh to start
-$SSH -o ConnectionAttempts=500 core@$IP id
+$SSH -o ConnectionAttempts=500 core@$API_VIP id
```

Comment:
Is it enough to just do 500 retries now that you dropped the while/sleep? the API will be resolved as soon as there's a record in libvirt, you need to wait for the bootstrap VM to take it over with keepalived.

Reply:
Yes, it looks it's enough, looking at the CI job here[1]:

```bash
Mar 27 23:18:16 |  + sudo systemctl reload NetworkManager
Mar 27 23:18:16 |  + ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5 -o ConnectionAttempts=500 core@192.168.111.5 id
Mar 27 23:21:46 |  Warning: Permanently added '192.168.111.5' (ECDSA) to the list of known hosts.
Mar 27 23:21:46 |  uid=1000(core) gid=1000(core) groups=1000(core),4(adm),10(wheel),16(sudo),190(systemd-journal) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
Mar 27 23:21:46 |  ++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5 core@192.168.111.5 '/usr/sbin/ip -o -4 a show eth0'
Mar 27 23:21:46 |  ++ awk '/brd/ {print $4}'
Mar 27 23:21:46 |  ++ awk -F / '{print $1}'
Mar 27 23:21:46 |  Warning: Permanently added '192.168.111.5' (ECDSA) to the list of known hosts.
Mar 27 23:21:47 |  + IP=192.168.111.45
```
[1] http://10.8.144.11:8080/job/dev-tools/404/consoleFull